.github/workflows/docker-publish-action.yml

```diff
@@ -1,19 +1,22 @@
-name: Publish API/UI Docker Image
-on: workflow_dispatch
+on:
+  workflow_call:
+    inputs:
+      IMAGE_NAME:
+        required: true
+        type: string
+      DOCKERFILE:
+        required: false
+        type: string
 
 env:
   REGISTRY: ghcr.io
-  IMAGE_NAME_UID2: iabtechlab/uid2-ssportal
 
 jobs:
   build-publish-docker:
     runs-on: ubuntu-latest
-    permissions:
-      contents: read
-      packages: write
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
```

Comment:
what does the version here actually mean? I see v2, v3 and v4 used throughout our workflows for ` actions/checkout`

Reply:
https://github.com/actions/checkout I just make sure it is uisng the latest version

Reply:
is it worth upgrading all of our workflows to use v4 then?

Reply:
I've upgraded all of them